redis/asyncio/cluster.py

```diff
@@ -298,6 +306,17 @@ def __init__(
             # Call our on_connect function to configure READONLY mode
             kwargs["redis_connect_func"] = self.on_connect
 
+        if retry or retry_on_error or connection_error_retry_attempts > 0:
```

Comment:
So in the async cluster (and only in the async code) we have this additional argument: 'connection_error_retry_attempts', which is mutually exclusive with 'retry'. 
Our guide line is to keep backward compatibility, and since the default value of [connection_error_retry_attempts is set to 5](https://github.com/redis/redis-py/blob/6b3e0b491c4a348ca2e7a332e6f3b23a5da3c461/redis/asyncio/cluster.py#L217), we need to try to maintain this default retry behavior. 
There are couple of options I thought of to maintain the same behavior, but none of them isn't great:
The first is to leave the default value as is and raise exceptions when mutually exclusive arguments passed together:
```
        if retry and retry_on_error:
            raise DataError("'retry' cannot be passed along with "
                            "'retry_on_error'. "
                            "To control these variables pass a "
                            "Retry object set with the required retries "
                            "and/or errors.")
        if retry and retry._retries != connection_error_retry_attempts:
            raise DataError("Retry attempts set in the passed 'retry' object do not match "
                            "'connection_error_retry_attempts'."
                            "'connection_error_retry_attempts' should be set to None or to "
                            "the number of retries set in the Retry object.")
```
However, as connection_error_retry_attempts == 5 by default, if someone will create a new async cluster instance and pass a retry object it will require him to set connection_error_retry_attempts to None/the same retries set in the retry object. e.g.:
```
r = RedisCluster(host, port, retry=Retry(backoff(), 4), connection_error_retry_attempts=None)
```
which obviously doesn't look good.
To solve it, we can change the connection_error_retry_attempts default value to None/0 by default, but then users that were used to get 5 retries will now have none. Do you think that it's something we can do if we'll announce about this change in the release notes?
Another option is to allow passing retry object with different retry attempts than in connection_error_retry_attempts and ignore the later when retry is passed. We can add a documentation of that, or even log a warning stating that we ignored connection_error_retry_attempts, but I don't think it would be readable at all for the user. 

WDYT? do you have a better option?

Reply:
Retry consistency (with the other interfaces as well) is generally what we should be aiming for. I think the entire retry piece is off - across all our interfaces.  In general, we should have a Retry object, which during initialization reqiures a number of retries - across the board. This will break everything, but IMHO is a worthy breakage (in the future).

For this release and these blocks...  I think that a variant of the if/else check is reasonable. Raising an exception when mutually exclusive arguments are passed in is fine - so long as doctstings and contributed examples indicate that this is the desired behaviour.